plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaFilterManager.java

```diff
@@ -91,30 +92,20 @@ public KafkaFilteringResult getKafkaFilterResult(
 
         if (!constraint.isAll()) {
             Set<Long> partitionIds = partitionInfos.stream().map(partitionInfo -> (long) partitionInfo.partition()).collect(toImmutableSet());
-            Optional<Range> offsetRanged = Optional.empty();
-            Optional<Range> offsetTimestampRanged = Optional.empty();
-            Set<Long> partitionIdsFiltered = partitionIds;
-            Optional<Map<ColumnHandle, Domain>> domains = constraint.getDomains();
 
-            for (Map.Entry<ColumnHandle, Domain> entry : domains.get().entrySet()) {
-                KafkaColumnHandle columnHandle = (KafkaColumnHandle) entry.getKey();
-                if (!columnHandle.isInternal()) {
-                    continue;
-                }
-                switch (columnHandle.getName()) {
-                    case PARTITION_OFFSET_FIELD:
-                        offsetRanged = filterRangeByDomain(entry.getValue());
-                        break;
-                    case PARTITION_ID_FIELD:
-                        partitionIdsFiltered = filterValuesByDomain(entry.getValue(), partitionIds);
-                        break;
-                    case OFFSET_TIMESTAMP_FIELD:
-                        offsetTimestampRanged = filterRangeByDomain(entry.getValue());
-                        break;
-                    default:
-                        break;
-                }
-            }
+            ImmutableMap<String, Domain> domains = constraint.getDomains().orElseThrow()
```

Comment:
Same question as https://github.com/trinodb/trino/pull/14224#discussion_r987677766

Reply:
cc: @losipiuk Do we have some convention about using interfaces (or superclasses) instead of concrete ones for local vars? I've received this comment in past from you as well - but probably we can configure Modernizer instead of manually reviewing and hoping we catch them?

Reply:
cc @ksobolew

Reply:
No sense to discuss here.